src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java

```diff
@@ -101,6 +115,27 @@ public void cannotUpdateNonexistentPatronNoticePolicy() throws InterruptedExcept
     MatcherAssert.assertThat(response.getStatusCode(), CoreMatchers.is(404));
   }
 
+  @Test
+  public void canGetAllPatronNoticePolicies() throws MalformedURLException, InterruptedException, ExecutionException,
+    TimeoutException {
+
+    PatronNoticePolicy entity = new PatronNoticePolicy();
+    entity.setName("firstPolicy");
+
+    createPatronNoticePolicy(entity);
+    createPatronNoticePolicy(entity.withName("secondPolicy"));
+
+    CompletableFuture<JsonResponse> getAllCompleted = new CompletableFuture<>();
+
+    client.get(patronNoticePoliciesStorageUrl(), StorageTestSuite.TENANT_ID, ResponseHandler.json(getAllCompleted));
+
+    JsonResponse response = getAllCompleted.get(5, TimeUnit.SECONDS);
+
+    MatcherAssert.assertThat("Failed to get all patron notice policies", response.getStatusCode(), CoreMatchers.is(200));
+    MatcherAssert.assertThat(response.getJson().getJsonArray("patronNoticePolicies").size(), CoreMatchers.is(2));
+    MatcherAssert.assertThat(response.getJson().getInteger("totalRecords"), CoreMatchers.is(2));
```

Comment:
There needs to be a check that the response actually contains the names "firstPolicy" and "secondPolicy"

Reply:
added name check

Reply:
Thanks.
This line is duplicate: `MatcherAssert.assertThat(policies.size(), CoreMatchers.is(2));`
There is no "sortBy" so the database may return the policies in any order. You can use a matcher that matches both names but in any order: http://hamcrest.org/JavaHamcrest/javadoc/2.0.0.0/org/hamcrest/collection/IsArrayContainingInAnyOrder.html

Reply:
Thank you. Rewrited the code using `arrayContainingInAnyOrder`.